### PR TITLE
Update dnd character exercise

### LINKED ISSUE #764: "D&D Character" requires using a class
The solution stub for the "D&D Character" exercise contains a class for the type to be implemented, and the unit tests directly call the constructor, which leaves the student no option than to actually solve this using a class. This is not preferred in F#, though, and there's no good reason why using a class would make sense here in any way.

The tests should just call a creation function here and not prescribe an implementation. I'm not sure what to do about the solution stub, though.

### PR DESCRIPTION
**Pull request by davidelettieri**
Close #764, I hope this is a satisfying implementation, if not I'll be happy to change whatever needed :)

I also updated another small detail of the generator since the test "Random ability is within range" was creating the value of ability once and testing in ten times, intead of creating it 10 times and testing once per value.

### CODE REVIEW

File: exercises/dnd-character/DndCharacterTest.fs [hunk 1]
```diff
@@ -73,15 +73,15 @@ let ``Ability modifier for score 18 is +4`` () =
 
 [<Fact(Skip = "Remove to run test")>]
 let ``Random ability is within range`` () =
-    let ability = ability()
     for i in 1 .. 10 do
+        let ability = ability()
```
Comment (robkeim) [inline exercises/dnd-character/DndCharacterTest.fs]: Nice catch here :+1:

Comment (ErikSchierboom) [reply exercises/dnd-character/DndCharacterTest.fs]: Indeed! Well caught!

File: exercises/dnd-character/DndCharacter.fs [hunk 2]
```diff
@@ -8,12 +8,7 @@ let modifier x =
...
-    member __.Strength with get() = failwith "You need to implement this function."
-    member __.Dexterity with get() = failwith "You need to implement this function."
-    member __.Constitution with get() = failwith "You need to implement this function."
-    member __.Intelligence with get() = failwith "You need to implement this function."
-    member __.Wisdom with get() = failwith "You need to implement this function."
-    member __.Charisma with get() = failwith "You need to implement this function."
-    member __.Hitpoints with get() = failwith "You need to implement this function."
+// TODO: define the Character type
 
+let mkCharacter() =
```
Comment (TeaDrivenDev) [inline exercises/dnd-character/DndCharacter.fs]: Maybe `createCharacter ()`? There's no reason to abbreviate things here.

**Comment by TeaDrivenDev** [comment]
Looks good, except for the function name that I'd change.

File: exercises/dnd-character/DndCharacter.fs (hunk 2 above)
Comment (davidelettieri) [reply exercises/dnd-character/DndCharacter.fs]: You're right, I'll update accordingly

**Comment by davidelettieri** [comment]
I renamed the function as requested

**Comment by robkeim** [comment]
Merging this given @TeaDrivenDev's comment has been addressed.

Thanks @davidelettieri for the contribution! :tada:

**Pull request merged**